Terminal.Gui/Views/Scroll/ScrollBar.cs

```diff
@@ -0,0 +1,118 @@
+﻿#nullable enable
+
+using System.ComponentModel;
+
+namespace Terminal.Gui;
+
+/// <summary>ScrollBars are views that display a 1-character scrollbar, either horizontal or vertical</summary>
+/// <remarks>
+///     <para>
+///         The scrollbar is drawn to be a representation of the Size, assuming that the scroll position is set at
+///         Position.
+///     </para>
+///     <para>If the region to display the scrollbar is larger than three characters, arrow indicators are drawn.</para>
+/// </remarks>
+public class ScrollBar : View
+{
+    /// <inheritdoc/>
+    public ScrollBar ()
+    {
+        _scroll = new (this);
+        _decrease = new (this);
+        _increase = new (this, VariationMode.Increase);
+        Add (_scroll, _decrease, _increase);
+
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
+
+        _scroll.PositionChanging += Scroll_PositionChanging;
+        _scroll.PositionChanged += Scroll_PositionChanged;
+        _scroll.SizeChanged += _scroll_SizeChanged;
+    }
+
+    private readonly Scroll _scroll;
```

Comment:
I know this isn't ready for review yet, but I took a look...

Why are these not just subviews? Why don't they just access `SuperView` to access "host"?

Reply:
`Scroll` and `ScrollBar` can be used separately. When only the `Scroll` is used the superview won't be a `ScrollBar` and thus the `Scroll` don't have `ScrollBar` as host. Am I wrong here?

Reply:
I'm very confused. 

TG has Subviews and SuperViews to support view containment. ScrollBar contains 3 subviews. Why are you not just adding them with Add instead of inventing a new containment system with "_host"?

Reply:
You've absolutely correct, sorry. I removed all the host parameter on the constructors. I renamed the `_host` to a private method `SupView` and maintained the _scroll, _decrease and _increase for convenience because they help on handling these objects. Fixed in https://github.com/gui-cs/Terminal.Gui/pull/3498/commits/ae3fcef17a44ef1797ab54c1e1ba73b8512bd813. Thanks.